src/bootstrap/flags.rs

```diff
@@ -503,6 +503,20 @@ Arguments:
             }
         };
 
+        if let Subcommand::Check { .. } = &cmd {
+            if matches.opt_str("stage").is_some() {
```

Comment:
I'm not sure how much values this gives but maybe erroring out on `stage > 0` and allowing `--stage 0` would be better?

Reply:
I'd rather just prevent it entirely, there's not really any reason to permit passing a zero I think -- users shouldn't treat x.py check as "having" stages in some sense.